pkg/queue/cluster_queue.go

```diff
@@ -89,17 +91,27 @@ func workloadKey(i *workload.Info) string {
 	return workload.Key(i.Obj)
 }
 
-func newClusterQueue(cq *kueue.ClusterQueue, wo workload.Ordering) (*ClusterQueue, error) {
-	cqImpl := newClusterQueueImpl(wo, realClock)
+func newClusterQueue(ctx context.Context, client client.Client, cq *kueue.ClusterQueue, wo workload.Ordering, fsConfig *config.AdmissionFairSharing) (*ClusterQueue, error) {
+	enableAdmissionFs, fsResWeights := isAdmissionFsEnabled(cq, fsConfig)
+	cqImpl := newClusterQueueImpl(ctx, client, wo, realClock, fsResWeights, enableAdmissionFs)
 	err := cqImpl.Update(cq)
 	if err != nil {
 		return nil, err
 	}
 	return cqImpl, nil
 }
 
-func newClusterQueueImpl(wo workload.Ordering, clock clock.Clock) *ClusterQueue {
-	lessFunc := queueOrderingFunc(wo)
+func isAdmissionFsEnabled(cq *kueue.ClusterQueue, fsConfig *config.AdmissionFairSharing) (bool, map[corev1.ResourceName]float64) {
```

Comment:
```suggestion
func afsResourceWeights(cq *kueue.ClusterQueue, fsConfig *config.AdmissionFairSharing) (bool, map[corev1.ResourceName]float64) {
```
if is a bit weird to call function "is" when returning more than bool. The bool probably should be the second argument, as more idiomatic (as golang returns "ok" bool as second when checking maps).

Reply:
Done